R/BUILDING.md

```diff
@@ -0,0 +1,13 @@
+## Building SparkR on Windows
```

Comment:
I will rename this to `WINDOWS.md` :) Basically I had to do a bunch of things to get it to build on windows and I decided to document it.